tests/validate_license_expr.rs

```diff
@@ -0,0 +1,72 @@
+extern crate license_exprs;
+
+use license_exprs::validate_license_expr;
+
+macro_rules! assert_licenses_ok {
+    ($($license:expr),+ $(,)+) => {{
+        $(assert!(validate_license_expr($license).is_ok());)+
+    }}
+}
+
+macro_rules! assert_licenses_err {
+    ($($license:expr),+ $(,)+) => {{
+        $(assert!(validate_license_expr($license).is_err());)+
+    }}
+}
+
+#[test]
+fn single_license() {
+    assert_licenses_ok! {
+        "MIT",
+        "MIT ",
+        " MIT",
+        " MIT ",
+        "    MIT    ",
+        "MIT+",
+    }
+}
+
+#[test]
+fn compound_license() {
+    assert_licenses_ok! {
+        "GPL-3.0+ WITH Classpath-exception-2.0 OR MIT AND AAL",
+        "MIT AND Apache-2.0",
+        "MIT OR Apache-2.0",
+        "GPL-3.0+ WITH Classpath-exception-2.0",
+        "MIT   AND    Apache-2.0",
+    }
+}
+
+#[test]
+fn fails_invalid_license() {
+    assert_licenses_err! {
+        "asdfghjkl",
+        "MIT AND qwerty",
+    }
+}
+
+#[test]
+fn fails_incorrect_structure() {
+    assert_licenses_err! {
+        "()",
+        "(MIT",
+        "MIT)",
+        "MIT Apache-2.0",
+        "MIT and Apache-2.0",
+        "MIT or Apache-2.0",
+        "GPL-3.0+ with Classpath-exception-2.0",
```

Comment:
There's some discussion about making operators case insensitive in spdx/spdx-spec#63, but [the 2.1 ABNF requires uppercase operators][1].  So I'm fine requiring failures here for now, but we may need to revisit these if the SPDX discussion results in them (or license expressions as a whole) becoming case insensitive.

[1]: https://spdx.org/spdx-specification-21-web-version#h.jxpfx0ykyb60